### PR TITLE
Use node model state to update the android notification

### PR DESCRIPTION
**Pull request by johnny9**
The AndroidNotifier class connects to the NodeModel's state signals and uses JNI to send callbacks to the Android service managing the foreground notification.

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/unsecure_win_gui.zip?branch=pull/331)
[Intel macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/unsecure_mac_gui.zip?branch=pull/331)
[Apple Silicon macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/unsecure_mac_arm64_gui.zip?branch=pull/331)
[ARM64 Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/unsecure_android_apk.zip?branch=pull/331)

### CODE REVIEW

**Comment by johnny9** [comment]
from[ 6e0d079 ](https://github.com/bitcoin-core/gui-qml/commit/6e0d079fbb3fdffbcb835b628a52f416b5deb6e5)to 1ef10e1
 - Fixed lint error
 - Made sure androidnotifier module is only compiled when targeting android using TARGET_ANDROID autotools conditional

**Comment by johnny9** [comment]
https://github.com/BitcoinDesign/Bitcoin-Core-App/issues/53

**Comment by johnny9** [comment]
from[ 1ef10e1 ](https://github.com/bitcoin-core/gui-qml/commit/1ef10e138486e74586799ecf0d8eef3f9499af78)to 90d250d
  - rebased with main

**Comment by johnny9** [comment]
Update from[ eb53a46 ](https://github.com/bitcoin-core/gui-qml/commit/eb53a46f4a7a1043d1b3cbff8d517d6d4ac1b3ba)to acd2d3e:
  - rebased with main

**Comment by johnny9** [comment]
update from[ acd2d3e ](https://github.com/bitcoin-core/gui-qml/commit/acd2d3e0ca55654afc03f1074e4c5d32d0473cfa)to b641df4:
  - fixed merge error on previous rebase. (#include qml/nodemodel.h to qml/models/nodemodel.h)

**Review by jarolrod — COMMENTED** [review]
Approach ACK

I've been learning more about what we really need to do here in order to interact with the Android and the JVM, and this code is looking good here. 

The androidnotifier.cpp file contains the importance pieces of magic. We're gaining access to the JVM and storing it. Then we're setting up the wiring between nodeModel updates to calls into our BitcoinQt Service to update the notification text.

@johnny9 are you able to reproduce this: https://github.com/bitcoin-core/gui-qml/pull/331/files#r1211573422

File: src/qml/androidnotifier.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+AndroidNotifier::AndroidNotifier(const NodeModel & node_model,
+                                 QObject * parent)
+: QObject(parent)
+, m_node_model(node_model)
+{
+    QObject::connect(&node_model, &NodeModel::blockTipHeightChanged,
+                     this, &AndroidNotifier::onBlockTipHeightChanged);
+    QObject::connect(&node_model, &NodeModel::numOutboundPeersChanged,
+                     this, &AndroidNotifier::onNumOutboundPeersChanged);
+    QObject::connect(&node_model, &NodeModel::pauseChanged,
```
[the remark this answers is not in the document]
Comment (johnny9) [reply src/qml/androidnotifier.cpp:51]: What state does it show? I'm having a hard time reproducing

[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qml/androidnotifier.cpp:51]: Well now I also can't reproduce, may have just lagged in showing the update for whatever reason android decided.

**Review by jarolrod — APPROVED** [review]
ACK b641df470c94eabb4cb2559c831a942b086bd3db

**Pull request merged**